optaplanner-examples/src/main/java/org/optaplanner/examples/nurserostering/optional/score/NurseRosteringConstraintProvider.java

```diff
@@ -0,0 +1,480 @@
+                .filter(minMaxContractLine -> minMaxContractLine.getContractLineType() == ContractLineType.TOTAL_ASSIGNMENTS &&
+                        minMaxContractLine.isEnabled())
+                .join(Employee.class,
+                        Joiners.equal(MinMaxContractLine::getContract, Employee::getContract))
+                .ifNotExists(ShiftAssignment.class,
+                        Joiners.equal((contractLine, employee) -> employee, ShiftAssignment::getEmployee))
+                .filter((contract, employee) -> contract.isViolated(0))
+                .penalize("org.optaplanner.examples.nurserostering.solver",
+                        "Minimum and maximum number of assignments (no assignments)", HardSoftScore.ONE_SOFT,
+                        (contract, employee) -> contract.getViolationAmount(0));
+    }
+
+    // Min/Max consecutive working days
+    // These Min/Max constraints are implemented as two constraints for consistency with DRL
+    Constraint minimumConsecutiveWorkingDays(ConstraintFactory constraintFactory) {
+        return getConsecutiveShifts(constraintFactory.from(MinMaxContractLine.class)
+                .filter(minMaxContractLine -> minMaxContractLine
+                        .getContractLineType() == ContractLineType.CONSECUTIVE_WORKING_DAYS &&
+                        minMaxContractLine.isMinimumEnabled()))
+                                .filter((employeeContractPair,
+                                        shiftList) -> shiftList.getLength() < employeeContractPair.getRight().getMinimumValue())
+                                .penalize("org.optaplanner.examples.nurserostering.solver", "minimumConsecutiveWorkingDays",
+                                        HardSoftScore.ONE_SOFT,
+                                        (employeeContractPair, shiftList) -> employeeContractPair.getRight()
+                                                .getViolationAmount(shiftList.getLength()));
+    }
+
+    Constraint maximumConsecutiveWorkingDays(ConstraintFactory constraintFactory) {
+        return getConsecutiveShifts(constraintFactory.from(MinMaxContractLine.class)
+                .filter(minMaxContractLine -> minMaxContractLine
+                        .getContractLineType() == ContractLineType.CONSECUTIVE_WORKING_DAYS &&
+                        minMaxContractLine.isMaximumEnabled()))
+                                .filter((employeeContractPair,
+                                        shiftList) -> shiftList.getLength() > employeeContractPair.getRight().getMaximumValue())
+                                .penalize("org.optaplanner.examples.nurserostering.solver", "maximumConsecutiveWorkingDays",
+                                        HardSoftScore.ONE_SOFT,
+                                        (employeeContractPair, shiftList) -> employeeContractPair.getRight()
+                                                .getViolationAmount(shiftList.getLength()));
```

Comment:
Please take care to maintain proper indentation, the filtrers and penalties ought to start at the same column. This is just one instance of multiple of this problem.